src/app/containers/CpsAssetPageMain/index.test.jsx

```diff
@@ -1,24 +1,37 @@
 import React from 'react';
-import { shouldMatchSnapshot } from '@bbc/psammead-test-helpers';
+import { StaticRouter } from 'react-router-dom';
+import matchSnapshot from './testHelpers';
 import { ServiceContextProvider } from '#contexts/ServiceContext';
 import { RequestContextProvider } from '#contexts/RequestContext';
 import CpsAssetPageMain from '.';
+import preprocessor from '#lib/utilities/preprocessor';
+import convertToOptimoBlocks from '#lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks';
 import pidginPageData from '#data/pidgin/cpsAssets/tori-49450859';
 
+const pageData = preprocessor(pidginPageData, [convertToOptimoBlocks]);
+
 describe('CpsAssetPageMain', () => {
-  shouldMatchSnapshot(
-    'should match snapshot',
-    <ServiceContextProvider service="pidgin">
-      <RequestContextProvider
-        bbcOrigin="https://www.test.bbc.co.uk"
-        isAmp={false}
-        pageType="MAP"
-        pathname="/pidgin/tori-49450859"
-        service="pidgin"
-        statusCode={200}
-      >
-        <CpsAssetPageMain service="pidgin" pageData={pidginPageData} />
-      </RequestContextProvider>
-    </ServiceContextProvider>,
-  );
+  it('should match snapshot', async done => {
+    const component = (
+      /*
+        for the value it would bring, it is much simpler to wrap a react-router Link in a Router, rather than mock a Router or pass some mocked context.
+      */
+      <StaticRouter>
+        <ServiceContextProvider service="pidgin">
+          <RequestContextProvider
+            bbcOrigin="https://www.test.bbc.co.uk"
+            isAmp={false}
+            pageType="MAP"
+            pathname="/pidgin/tori-49450859"
+            service="pidgin"
+            statusCode={200}
+          >
+            <CpsAssetPageMain service="pidgin" pageData={await pageData} />
```

Comment:
Should this await be here?

Reply:
`CpsAssetPageMain` expects to receive the `pageData` as an object rather than a Promise, so I believe this await is required

Reply:
Just looks odd awaiting a variable that suggests it's data. You could move line 11 to onside the it block and await it like this `const pageData = await preprocessor(pidginPageData, [convertToOptimoBlocks]);`

Reply:
Yeah good point it is hard to read. I've moved the const and await into the function and also stopped setting the component const, should make it the test more readable